### PR TITLE
Fixed-wing position control: some incremental clean up of the class

### PR DESCRIPTION
**Pull request by tstastny**
**Describe problem solved by this pull request**
The current state of the fixed-wing position control module is quite dismal. It's quite hard to understand what is happening, and there is very little documentation to help readers along the way. There are also quite some artifacts and mistakenly placed logic throughout specifically because the class is hard to read.

**Describe your solution**
This PR follows #19778. Tackles the second part in #19595.

**What's done?**
- documentation of units, params, returns, and descriptions for variables and methods
- rename ambiguous or erroneous state variables
- remove unused or unnecessary input arguments to functions
- remove ugly header white space

I named things what I thought was clearest. If there are disagreements, please let me know. Also for method documentation, please check and make sure I didn't make any unit or intention mistakes.

### CODE REVIEW

**Review by Jaeyoung-Lim — APPROVED** [review]

**Comment by tstastny** [comment]
@Jaeyoung-Lim @dagar @sfuhrer i rebased on latest master (one small is_vtol naming change)

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp [hunk 1]
```diff
@@ -362,74 +422,193 @@ class FixedwingPositionControl final : public ModuleBase<FixedwingPositionContro
...
+	 * @param ground_speed Local 2D ground speed of vehicle [m/s]
+	 * @param pos_sp_prev previous position setpoint
+	 * @param pos_sp_curr current position setpoint
+	 * @param pos_sp_next next position setpoint
+	 */
+	void control_auto(const float control_interval, const Vector2d &curr_pos, const Vector2f &ground_speed,
+			  const position_setpoint_s &pos_sp_prev, const position_setpoint_s &pos_sp_curr, const position_setpoint_s &pos_sp_next);
+
+	/**
+	 * @brief Controls altitude and airspeed for a fixed-bank loiter.
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp:453]: Could add that this is meant as a failsafe mode after lateral position estimation failure.

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp [hunk 2]
```diff
@@ -362,74 +422,193 @@ class FixedwingPositionControl final : public ModuleBase<FixedwingPositionContro
...
+	 *
+	 * @param control_interval Time since last position control call [s]
+	 */
+	void control_auto_fixed_bank_alt_hold(const float control_interval);
+
+	/**
+	 * @brief Control airspeed with a fixed descent rate and roll angle.
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp:460]: Also here add note about it being a failsafe mode

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp [hunk 3]
```diff
@@ -362,74 +422,193 @@ class FixedwingPositionControl final : public ModuleBase<FixedwingPositionContro
...
+	 *
+	 * @param pos_sp_curr Current position setpoint
+	 * @return Adjusted position setpoint type
+	 */
+	uint8_t	handle_setpoint_type(const position_setpoint_s &pos_sp_curr);
+
+	/* automatic control methods */
+
+	/**
+	 * @brief Position control for all automatic modes except takeoff and landing
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp:440]: (+the failsafe modes control_auto_fixed_bank_alt_hold() and control_auto_descend())

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp [hunk 4]
```diff
@@ -362,74 +422,193 @@ class FixedwingPositionControl final : public ModuleBase<FixedwingPositionContro
...
+	 * @brief A wrapper function to call the TECS implementation
+	 *
+	 * @param control_interval Time since the last position control update [s]
+	 * @param alt_sp Altitude setpoint, AMSL [m]
+	 * @param airspeed_sp Calibrated airspeed setpoint [m/s]
+	 * @param pitch_min_rad Nominal pitch angle command minimum [rad]
+	 * @param pitch_max_rad Nominal pitch angle command maximum [rad]
+	 * @param throttle_min Minimum throttle command [0,1]
+	 * @param throttle_max Maximum throttle command [0,1]
+	 * @param throttle_cruise Cruise throttle command [0,1]
```
Comment (sfuhrer) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.hpp:607]: It's not really a command but rather a TECS setting no? So something like "Configured throttle for level flight"

**Review by sfuhrer — DISMISSED** [review]
Only a very little amount of nitpicking from my side, otherwise great direction and I found nothing that is off.

**Review by Jaeyoung-Lim — APPROVED** [review]

**Review by sfuhrer — APPROVED** [review]

**Pull request merged**